core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -47,7 +48,11 @@ public class PartitionsTable extends BaseMetadataTable {
             Types.NestedField.required(1, "partition", Partitioning.partitionType(table)),
             Types.NestedField.required(2, "record_count", Types.LongType.get()),
             Types.NestedField.required(3, "file_count", Types.IntegerType.get()),
-            Types.NestedField.required(4, "spec_id", Types.IntegerType.get()));
```

Comment:
In the past we have general said that readers should not rely on position of fields in tables, only on names. That said every time we do something like this I think we end up breaking somebody. 

If we are gonna move things it probably should go before file and record count, though rather than after all the stats.

Reply:
Ok. Thanks. I think we can move spec_id to field id = 2 (before file and record count) 

@szehon-ho : What is your opinion on this?

Reply:
@RussellSpitzer I think it was , we should keep ids/names but not necessarily order.  Ref: https://github.com/apache/iceberg/pull/3015#issuecomment-951133690

Didn't see we broke anything that time, we resolved https://github.com/apache/iceberg/pull/3015#discussion_r753631713 eventually as a classpath issue with old classes still in play.

Reply:
So being said, we should not change the id of spec_id here, but ok to move it to after partition as you guys are saying.

Reply:
Thanks for linking the previous discussions. I will keep the name and id as it is while moving the field